### PR TITLE
Tuples: adding tests and work notes

### PR DESCRIPTION
**Pull request by jcouv**
Adding a initial work item list.
Adding some unittests. 

@dotnet/roslyn-compiler for review.

### CODE REVIEW

File: docs/features/tuples.work.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+- [ ] Generating and loading metadata for user-defined member names
+- [ ] Figure out full behavior for reserved member names
+- [ ] Support tuples 8+
+- [ ] Interop with System.Tuple, KeyValuePair
+- [ ] Publish short-term nuget package for tuples library
+- [ ] Get tuples library into corefx
+- [ ] Debugger bugs
+    - [ ] Tuple debug display is {(1, 2)} because ValueTuple.ToString() returns "(1, 2)"
+    - [ ] Tuple-returning method declaration shows values if names match local variables
+
```
Comment (VSadov) [inline docs/features/tuples.work.md]: target typing
OHI validation / fixing
control/data flow - mostly testing.

Comment (VSadov) [inline docs/features/tuples.work.md]: semantic info and other IDE stuff
debugger / watch window / expr evaluator./ EnC
various expression contexts 
  expr trees
  dynamic
  unsafe code / pointers
  optional parameters constants 
  xml docs

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs [hunk 2]
```diff
@@ -867,6 +867,95 @@ static void Main()
...
+        System.Console.WriteLine($""{x.first} {x.second}"");
+    }
+
+    (T1 first, T2 second) M<T1, T2>()
+    {
+        return (default(T1), default(T2));
+    }
+}
+";
+            // PROTOTYPE this should not throw
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs]: If it does throw, please skip the test.
PROTOTYPE comments should start with text of the form
```cs
// PROTOTYPE (tuples): 
```
please

**Comment by gafter** [comment]
(Done with this review pass)

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs (hunk 2 above)
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs]: I'll change to PROTOTYPE(tuples).

The test itself doesn't throw, but `CompileAndVerify` does (the exception is caught by `Assert.Throws`. Is that ok?

Comment (gafter) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs]: If CompileAndVerify is not supposed to throw an exception, but it does, then I think you should skip the test with a bug to fix it.

File: docs/features/tuples.work.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+Tuples Work Items
```
Comment (gafter) [inline docs/features/tuples.work.md]: Can you please add a *specification* for the tuple feature as `tuples.md`?

Comment (jcouv) [inline docs/features/tuples.work.md]: Yes. Let's bring that up with Vlad.

**Comment by jcouv** [comment]
@gafter Latest commit addresses your feedback. I went ahead and fixed PROTOTYPE(tuples) everywhere.
@VSadov for a second review. 
Thanks!

File: docs/features/tuples.work.md (hunk 3 above)
Comment (VSadov) [inline docs/features/tuples.work.md]: Right. Now that we have settled on target typing details we know most of the parts, so we can start specking it.
That can go in a separate change, I assume.

**Comment by VSadov** [comment]
LGTM

**Comment by gafter** [comment]
LGTM

**Comment by jcouv** [comment]
@dotnet-bot retest prtest/win/dbg/unit64 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit64/5709/
// Retest reason: file used by another process

**Pull request merged**